### PR TITLE
add virulencefinder 2.0.4

### PR DESCRIPTION
**Pull request by kapsakcjtest**
~~_Setting as a draft for now, just a few small changes to make first_~~

Adding a new tool to this repo, VirulenceFinder 2.0.4 which can be used for detecting virulence genes in E. coli, Enterococcus, Staphylococcus aureus, & Listeria. It takes either assemblies or reads as input

The dockerfile largely mirrors the `serotypefinder/2.0.1/Dockerfile` as they have identical dependencies except for the database. Just changed around the `test` stage a bit.

Side note: I wrote and tested this dockerfile in GitPod as prep for the NE BRR docker training. These were the steps I took:

- fork repo to my personal account `kapsakcjtest/docker-builds`
- start GitPod workspace using my forked repo as the context URL
- write dockerfile, test code, commit changes in local git repo
- grant GitPod permissions to my GitHub account:
Untitled (2)
- push commits back to `master` branch in my forked repo `kapsakcjtest/docker-builds`
- create PR via GitHub web UI

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: virulencefinder/2.0.4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# set final working directory for production docker image (app layer only)
+WORKDIR /data
+
+### START OF TEST STAGE ###
+FROM app as test
+
+# set working directory for test layer
+WORKDIR /test
+
```
Comment (kevinlibuit) [inline virulencefinder/2.0.4/Dockerfile:80]: @kapsakcj thoughts on implementing a CMD layer?

**Comment by kapsakcjtest** [comment]
I pushed the image to my personal dockerhub repo for testing: https://hub.docker.com/r/kapsakcj/virulencefinder/tags

 image name & tag `kapsakcj/virulencefinder:2.0.4`

I added the CMD layer and it appears to work as intended:
```
$ docker run kapsakcj/virulencefinder:2.0.4
Unable to find image 'kapsakcj/virulencefinder:2.0.4' locally
2.0.4: Pulling from kapsakcj/virulencefinder
b549f31133a9: Pull complete
f03f52846eb6: Pull complete
c2fb552889e7: Pull complete
7fccaa5a00f7: Pull complete
c90e7b1843c7: Pull complete
57bd34485fbc: Pull complete
4f4fb700ef54: Pull complete
baf1ff929297: Pull complete
c32b0731f20f: Pull complete
5bafbe7bfd85: Pull complete
30ea6ad63210: Pull complete
Digest: sha256:3f2617fe6b35253a6255421e4044da8e0e2af9c6601ba3ce3cb43d63939e0eae
Status: Downloaded newer image for kapsakcj/virulencefinder:2.0.4
usage: virulencefinder.py [-h] -i INFILE [INFILE ...] [-o OUTDIR]
                          [-tmp TMP_DIR] [-mp METHOD_PATH] [-p DB_PATH]
                          [-d DATABASES] [-l MIN_COV] [-t THRESHOLD] [-x] [-q]

optional arguments:
  -h, --help            show this help message and exit
  -i INFILE [INFILE ...], --infile INFILE [INFILE ...]
                        FASTA or FASTQ input files.
  -o OUTDIR, --outputPath OUTDIR
                        Path to blast output
  -tmp TMP_DIR, --tmp_dir TMP_DIR
                        Temporary directory for storage of the results from
                        the external software.
  -mp METHOD_PATH, --methodPath METHOD_PATH
                        Path to method to use (kma or blastn)
  -p DB_PATH, --databasePath DB_PATH
                        Path to the databases
  -d DATABASES, --databases DATABASES
                        Databases chosen to search in - if non is specified
                        all is used
  -l MIN_COV, --mincov MIN_COV
                        Minimum coverage
  -t THRESHOLD, --threshold THRESHOLD
                        Minimum hreshold for identity
  -x, --extented_output
                        Give extented output with allignment files, template
                        and query hits in fasta and a tab seperated file with
                        gene profile results
  -q, --quiet

```

File: virulencefinder/2.0.4/README.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  - **Default % Identity threshold: 90%**. Adjust with `-t 0.95`
+  - **Default % coverage threshold: 60%**. Adjust with `-l 0.70`
+  - Use the `-x` flag (extended output) if you want the traditional/legacy VirulenceFinder output files `results_tab.tsv  results.txt  Virulence_genes.fsa  Hit_in_genome_seq.fsa`. Otherwise you will need to parse the default output file `data.json` for results
+  - (*RECOMMENDED*) Use raw reads due to the increased sensitivity (without loss of specificity) and the additional information gleaned from KMA output (specifically the depth metric). You also save time from having to assemble the genome first. [CITATION NEEDED, PROBABLY THE KMA PAPER]
+- Querying reads:
+  - This will run VirulenceFinder with `kma` (instead of ncbi-blast+)
+  - Only one of the PE read files is necessary. There is likely little benefit to using both R1 and R2. It will take longer to run if you use both R1 and R2 files.
+- Querying assemblies:
+  - This will run VirulenceFinder with `ncbi-blast+`
+  - VirulenceFinder does not clean up after itself. `tmp/` (which contains 7 different `.xml` files) will exist in the specified output directory
```
Comment (erinyoung) [inline virulencefinder/2.0.4/README.md:75]: I like this section in the readme

**Comment by kevinlibuit** [comment]
Looks good and builds locally without issue on my end. I do also really like the feel of things with the `CMD` layer defined.

**Review by kevinlibuit — APPROVED** [review]

**Pull request merged**